numpy/core/src/private/ufunc_override.c

```diff
@@ -32,8 +32,8 @@ has_non_default_array_ufunc(PyObject *obj)
                                                      "__array_ufunc__");
     }
 
-    /* Fast return for ndarray */
-    if ((PyObject *)Py_TYPE(obj) == ndarray) {
+    /* Fast return for ndarray and basic types */
+    if ((PyObject *)Py_TYPE(obj) == ndarray || _is_basic_python_type(obj)) {
```

Comment:
the following `PyArray_GetAttrString_SuppressException` (which we should remove as it does nothing for modern python) also checks the basic type but we call it on the type object not the object.
I don't understand why that is done, but stuff fails if I change it, a better comment would be good.